src/targets/gpu/mlir.cpp

```diff
@@ -1096,6 +1097,46 @@ std::string dump_mlir(module m, const std::vector<shape>& inputs)
     return mlir_print(&mlirOperationPrint, mod_op);
 }
 
+static std::string compute_dump_name(const module& m, const std::string& ext)
+{
+    std::vector<instruction_ref> sizes;
+    for(auto ins : iterator_for(m))
+    {
+        if(not contains({"convolution", "dot"}, ins->name()))
+            continue;
+        sizes.insert(sizes.end(), ins->inputs().begin(), ins->inputs().end());
+    }
+    auto name =
+        mlir_program::get_symbol_name(m) + "_" + shape::to_sizes_string(to_shapes(sizes)) + ext;
+    replace_string_inplace(name, ", ", "_");
+    replace_string_inplace(name, ":", "s");
+    return name;
+}
+
+void dump_mlir_to_file(module m, const std::vector<shape>& inputs, const fs::path& location)
+{
+    static std::mutex mutex;
+    const std::lock_guard<std::mutex> lock(mutex);
+
+    if(not inputs.empty())
+    {
+        adjust_param_shapes(m, inputs);
+    }
+    rewrite_reduce(m);
+
+    auto name = compute_dump_name(m, ".mlir");
+    std::cout << "Dumping MLIR file to: " << name << std::endl;
+    auto f    = location / name;
+
+    mlir_program mp;
+    mp.parse(m);
+    auto mod_op = mlirModuleGetOperation(mp.mmodule.get());
+
+    std::string mlir_str = mlir_print(&mlirOperationPrint, mod_op);
+
+    write_string(f, mlir_str);
```

Comment:
It looks like `write_string` in src/file_buffer.cpp doesn't account for exceptions. But that functionality should be added there rather than here, I believe. I can open a seperate PR. @pfultz2

Reply:
Yes, please, that would be a good idea. But regardless, you might still have to catch the exception here, in that case. Thanks.

May I suggest that you could please print the message to `cout` after calling `write_string()`, and print `f`, rather than `name`. This way if only it is successful do we get the printed message -- with the full path in it.

Reply:
And would the '/' in the file path work for Windows? (Repeat question, I added that somewhere else too).

Reply:
I think it would be better idea to have exception handling in write_string() so it applies to all instances of it.

Reply:
Dont add exception handling.